frontend-ssr/redux/exercise-submissions/submissions.reducer.ts

```diff
@@ -0,0 +1,49 @@
+import { SubmissionState } from './types';
+
+const initialState: SubmissionState = {
+  submissions: [],
+};
+export const submissionReducer = (
+  state = initialState,
+  action: { type: string; payload: any; id: string},
+): SubmissionState => {
+  switch (action.type) {
+    case 'GET_SUBMISSIONS':
+      return {
+        ...state.submissions,
+        submissions: action.payload,
+      };
+    case 'ADD_SUBMISSION':
+      return {
+        submissions: [...state.submissions, action.payload],
+      };
+    case 'REMOVE_SUBMISSION':
+      return {
+        submissions: state.submissions.filter((submission) => submission.id !== action.payload),
+      };
+    case 'EDIT_SUBMISSION':
+      return {
+        submissions: state.submissions.map((submission) => ((submission.id === action.payload)
+          ? { ...submission, editing: !submission.editing }
+          : submission)),
+      };
+    case 'UPDATE_SUBMISSION':
+      return {
+        submissions: state.submissions.map((submission) => {
+          if (submission.id === action.id) {
+            return {
+              ...submission,
+              username: action.payload.newUsername,
```

Comment:
The properties of `action.payload` should be the same as the submission. It's easier if we mantain the same interface, instead of having both `username`, `newUsername`, etc... Also, you shouldn't check for each property in particular. Imagine if our data structure grows and we'll have 10+ props :( 

So let's do

```
return { 
  ...submission,
  ...action.payload.submission
}
```